src/yamlprocessor/datapreprocessor.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+"""The datapreprocessor looks for the DIRECT_INCLUDE= keyword in the input
+   yaml and concatenates the associated file at this point in the input
+   file. The result is written to the output file or standard out if - is
+   specified.
+
+Example usage:
+    python datapreprocessor.py -o <output file> <input file> \
+        --define JOPA_AUX=/path/to/my/file
+    python datapreprocessor.py -o- <input file> \
+        --define JOPA_AUX=/path/to/my/file
+"""
+
+import argparse
+import sys
+
+
+class DataPreProcessor:
+
+    def __init__(self):
+        self.replacements = {}
+
+    def add_replacements_map(self, replacements):
+        self.replacements = replacements
+
+    def process_yaml(self, in_yaml, out_yaml):
+        # read yaml file
+        src_file = open(in_yaml, 'r')
+        lines = src_file.readlines()
+        src_file.close()
+
+        # process yaml file
+        new_line = []
+        for iline in lines:
+            # look for specific pattern in each line
+            if 'DIRECT_INCLUDE=' in iline:
+                # retrieve header file
+                yaml_header_File = iline.split('=')[1].rstrip()
+                # replace variables in the string
+                for key, value in self.replacements.items():
+                    yaml_header_File = \
+                        yaml_header_File.replace(f'${key}', value)
+                # open header file
+                with open(yaml_header_File, 'r') as file:
+                    auxFileData = file.read()
+                # update lines for new file
+                new_line.append(auxFileData)
+            else:
+                new_line.append(iline)
+        # save the result
+        if out_yaml == '-':
+            out_file = sys.stdout
+        else:
+            out_file = open(out_yaml, 'w')
+        out_file.writelines(new_line)
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Process input and output "
+                    "files with multiple --define options."
+    )
+
+    # Positional argument for input
+    parser.add_argument('input_file', type=str, help='Input file')
```

Comment:
Should this be:
`parser.add_argument('--input_file', '-i', type=str, help='Input file')`

Reply:
This is always a required input and the current method follows the same method as yp-data.